core/services/relay/evm/relayer_extender.go

```diff
@@ -64,12 +68,13 @@ func NewLegacyChains(ctx context.Context, opts legacyevm.ChainRelayOpts) (result
 		}
 
 		privOpts.Logger.Infow(fmt.Sprintf("Loading chain %s", cid), "evmChainID", cid)
-		chain, err2 := legacyevm.NewTOMLChain(ctx, enabled[i], privOpts)
+		chain, err2 := legacyevm.NewTOMLChain(ctx, enabled[i], privOpts, clientsByChainID)
 		if err2 != nil {
 			err = multierr.Combine(err, fmt.Errorf("failed to create chain %s: %w", cid, err2))
 			continue
 		}
 
+		clientsByChainID[cid] = chain.Client()
```

Comment:
Don't think I saw us test anywhere if this map is being built as expected based on configs. Is there anywhere we can add a test for this?

Reply:
I don't see a easy way to test this since it's just an argument passing down to the function. Maybe leave it for now and add test coverage when it's actually used in `NewL1GasOracle`